### PR TITLE
Add Debian automation

### PR DESCRIPTION
**Pull request by marcwrobel**
See #42.

### CODE REVIEW

**Comment by marcwrobel** [comment]
@captn3m0 , it seems to be an issue with the `/src/palo-alto-networks.py`. See https://github.com/endoflife-date/release-data/actions/runs/3706004931/jobs/6282166146 :

```
Traceback (most recent call last):
  File "/home/runner/work/release-data/release-data/src/palo-alto-networks.py", line 45, in <module>
    update_releases(html_id, ID_MAPPING[html_id])
  File "/home/runner/work/release-data/release-data/src/palo-alto-networks.py", line 35, in update_releases
    date = datetime.datetime.strptime(td_list[1].get_text(), "%B %d, %Y")
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "/opt/hostedtoolcache/Python/3.10.9/x64/lib/python3.10/_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data 'September 3rd, 2014' does not match format '%B %d, %Y'
```

**Comment by captn3m0** [comment]
PAN added some old dates with a new format on the website. Fixed: https://github.com/endoflife-date/release-data/commit/cc624a7cd4e3b0f69546be8b48101c80fc1e49f0

**Comment by marcwrobel** [comment]
Sure.

But does it mean we plan to add EOLed version 2 to 5 to https://endoflife.date/debian ?

**Comment by captn3m0** [comment]
Maybe - we can always add those later once we have the ability to hide the rows.

Plus, I look at this project (release-data) as being helpful elsewhere - having a canonical list of all debian versions with release dates in JSON format is definitely gonna be useful.

**Comment by marcwrobel** [comment]
> The closest is Wikipedia (https://en.wikipedia.org/wiki/Debian_version_history) today, but it's not machine-readable.

That would not be easy but feasible: the page looks structured enough. But I do not think it is worth it.

**Comment by marcwrobel** [comment]
> Maybe - we can always add those later once we have the ability to hide the rows.

I think `latest.py` will issue warnings about those versions. Am I correct ?

**Comment by captn3m0** [comment]
It only warns about releases that were issued in the last 30 days: https://github.com/endoflife-date/endoflife.date/blob/56fec14ebe6906172b1ee46afeff381a6b12b441/_auto/latest.py#L180-L189.

**Comment by marcwrobel** [comment]
Something similar with grep (much more common than ag / one less dependency to install on the CI) : `grep -Rh -B 10 '<define-tag revision>' . | grep -Eo '(release_date>(.*)<|revision>(.*)<)' | cut -d '>' -f 2,4 | tr -d '<' | paste - -`.

`git clone https://salsa.debian.org/webmaster-team/webwml.git --depth 1` seems faster than the sparse checkout. Maybe we can combine both ?

**Comment by captn3m0** [comment]
Yes. I did a shallow clone as well, forgot to add it on the comment.

**Comment by marcwrobel** [comment]
@captn3m0, I switched to the https://salsa.debian.org/webmaster-team/webwml/-/tree/master/english/News source. Ready for a review.

I stored the repository in ~/.cache, if I understood correctly it will be cache along with other repositories.

File: src/debian.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    call(f"git {git_opts} config core.sparseCheckout true", shell=True)
+    with open(f"{REPO_DIR}/.git/info/sparse-checkout", "w") as f:
+        f.write("english/News/")
+
+    ret_code = call(f"git {git_opts} pull --depth 1 origin master", shell=True)
+    exit(-ret_code) if ret_code < 0 else None
+
+
+def extract_releases():
+    child = subprocess.Popen(
```
Comment (captn3m0) [inline src/debian.py:35]: Would have preferred if this were in Python instead. But if this feels stable enough, we can merge.

Comment (marcwrobel) [reply src/debian.py:35]: I think it is.

I thought about it but reproducing all that commands in Python looked harder ;).

**Pull request merged**